CI is failing using the Counter Basic API test using the Board da1469x_dk_pro.

**Describe the bug**
CI is failing counter_basic_api when using -b da1469x_dk_pro

**To Reproduce**
 west build -p always -b da1469x_dk_pro ./zephyr/tests/drivers/counter/counter_basic_api/da1469x_dk_pro

**Expected behavior**
The platform build successfully.

**Impact**
CI is blocking this counter based PR such as -> https://github.com/zephyrproject-rtos/zephyr/pull/73236

**Logs and console output**
Build output:
/home/emilio/zephyrproject/zephyr/drivers/counter/counter_smartbond_timer.c:109:1: warning: control reaches end of non-void function [-Wreturn-type]
  109 | }
      | ^
[139/142] Generating isr_tables.c, isr_tables_vt.ld, isr_tables_swi.ld
FAILED: zephyr/isr_tables.c zephyr/isr_tables_vt.ld zephyr/isr_tables_swi.ld /home/emilio/zephyrproject/build/zephyr/isr_tables.c /home/emilio/zephyrproject/build/zephyr/isr_tables_vt.ld /home/emilio/zephyrproject/build/zephyr/isr_tables_swi.ld
cd /home/emilio/zephyrproject/build/zephyr && /home/emilio/zephyrproject/.venv/bin/python3 /home/emilio/zephyrproject/zephyr/scripts/build/gen_isr_tables.py --output-source isr_tables.c --linker-output-files isr_tables_vt.ld isr_tables_swi.ld --kernel /home/emilio/zephyrproject/build/zephyr/zephyr_pre0.elf --intlist-section .intList --intlist-section intList --sw-isr-table --vector-table
gen_isr_tables.py: error: multiple registrations at table_index 17 for irq 17 (0x11)
Existing handler 0x16006d09, new handler 0x16007ed5
Has IRQ_CONNECT or IRQ_DIRECT_CONNECT accidentally been invoked on the same irq multiple times?


**Environment (please complete the following information):**
 - OS: Linux 
 - Commit SHA: 4fc6506a8af33e8025e9119fdddc96e007acb3ca

**Additional context**
Will add results of gitbisect soon.

--- a/drivers/counter/counter_smartbond_timer.c
+++ b/drivers/counter/counter_smartbond_timer.c
@@ -105,6 +105,8 @@ static uint8_t counter_smartbond_pdc_trigger_get(const struct device *dev)
 		return MCU_PDC_TRIGGER_TIMER3;
 	case (uint32_t)TIMER4:
 		return MCU_PDC_TRIGGER_TIMER4;
+	default:
+		return 0;
 	}
 }
 

--- a/soc/renesas/smartbond/da1469x/Kconfig.defconfig
+++ b/soc/renesas/smartbond/da1469x/Kconfig.defconfig
@@ -4,7 +4,7 @@
 if SOC_SERIES_DA1469X
 
 config SMARTBOND_TIMER
-	default y if PM
+	default y if PM && !$(dt_nodelabel_enabled,timer2)
 
 config CORTEX_M_SYSTICK
 	default n if SMARTBOND_TIMER

